File: nowcoder/contest/23106/H.py
from bisect import bisect_left, bisect_right
from collections import Counter, defaultdict
import sys
def get_ints(): return map(int, sys.stdin.readline().strip().split())


[n] = get_ints()
a = list(get_ints())
for i in range(n):
    a[i] -= 500
a.sort()
ans = 0
for num in a:
    i = bisect_left(a, -num)
    ans += i*(-num)
    ans += (n-i)*num
    ans += abs(num)
print(ans)

File: nowcoder/contest/23106/H.cpp
#include <algorithm>
#include <iostream>
using namespace std;
const int MAXN = (1e6) + 1;
int a[MAXN];
int main() {
    int n;
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> a[i];
        a[i] -= 500;
    }
    sort(a, a + n);
    long long ans = 0;
    int tmp;
    for (int i = 0; i < n; i++) {
        tmp = lower_bound(a, a + n, -a[i]) - a;
        ans += tmp * (-a[i]);
        ans += (n - tmp) * a[i];
        ans += abs(a[i]);
    }
    cout << ans << endl;
}